367 netlify cra frontend

nlkluth opened the pull request:
This PR closes #367 

#### What does this PR do?
Kicking off this PR though there are still a few things left todo:
- [x] validate on heroku that API still works as standalone (gonna use the heroku test env for this)
- [x] figure out where we want netlify to live
- [x] ~~do some sort of migration with DNS switchover?~~ set up a migration path (keeping the front-end in heroku so we can use that until we can switch the DNS)
- [x] separates front end from backend
- [x] moves project into `create-react-app` template
- [x] Moves front end to Netlify

#### How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

![](https://media.giphy.com/media/3osxYaDlxLcexsQiKk/giphy.gif)

wdoug committed:
Add actual terms! Woo!

wdoug committed:
Don't fail tests on minor security risks for now

nlkluth committed:
wip create-react-app frontend

nlkluth committed:
wip create-react-app frontend

nlkluth committed:
moving files

nlkluth committed:
moving files

nlkluth committed:
moving server packages and removing front-end boot

nlkluth committed:
moving server packages and removing front-end boot

nlkluth committed:
Merge branch '367-netlify-cra-frontend' of https://github.com/codefordenver/encorelink into 367-netlify-cra-frontend

Conflicts:
	src/components/Terms.js
	src/components/__tests__/__snapshots__/Terms.spec.js.snap
	src/stories/__tests__/storyshots.spec.js

nlkluth committed:
update package-lock

nlkluth committed:
fixing issues that I caused :blush:

nlkluth committed:
updating terms to what it should be

nlkluth committed:
remove css link as react-scripts binds it up

nlkluth committed:
moved stories over

nlkluth committed:
remove build, add cpy for 404.html

nlkluth committed:
gitignore build

nlkluth committed:
moving files over, working on storybook

nlkluth committed:
update eslint to match create-react-app

nlkluth committed:
fix tests, use updated storybook & addons

nlkluth committed:
start server at same time as front end for local dev

nlkluth commented:
Looks like heroku pukes; `heroku local:start` doesn't complain, although it doesn't like that there is no index route

wdoug commented:
Nice gif @nlkluth

wdoug commented on src/scss/app.css:
```diff
@@ -0,0 +1,3423 @@
+@charset "UTF-8";
```
These built files should be gitignored

fredmon3 committed:
Remove css files

fredmon3 committed:
add css to gitignore

wdoug commented on package.json:
```diff
@@ -3,100 +3,54 @@
+    "test": "react-scripts test --env=jsdom",
+    "eject": "react-scripts eject",
+    "build-css": "node-sass-chokidar src/ -o src/",
+    "storybook": "start-storybook -p 9001 -c .storybook",
```
Let's add a `postinstall` script to automatically install the server dependencies after `npm install`

wdoug replied:
The script can be `npm install --prefix server`

wdoug commented:
I am getting an error when trying to run react storybook:
```
ERROR in ./src/scss/app.scss
Module parse failed: Unexpected character '@' (1:0)
You may need an appropriate loader to handle this file type.
| @import './vendors/foundation/foundation';
| @include foundation-everything(true);
```

wdoug committed:
Merge remote-tracking branch 'origin/master' into 367-netlify-cra-frontend

wdoug commented on server/config.json:
```diff
@@ -1,7 +1,7 @@
   "restApiRoot": "/api",
   "host": "0.0.0.0",
-  "port": 3000,
+  "port": 54321,
```
@nlkluth is this just arbitrary?

wdoug replied:
@jsullivan5 this file is configuring the port we are listening for the loopback server. My guess is that it was just changed to an arbitrary number since 3000 is the default for create-react-app (the new front end setup).

nlkluth replied:
Yeah - Can switch it to whatever though!

jsullivan5 commented:
@wdoug I'm not too familiar with this loopback framework, but it handles a lot of
the database endpoints for you.  Under the hood it must be routing the
database calls through /api.  The dev port is 3000 and 54321 could be
postgres. The host looks like its set to a generic default.   Not certain
though.  Documentation: https://loopback.io/doc/en/lb2/config.json.html

On Wed, Apr 4, 2018 at 8:57 PM, Will Douglas <notifications@github.com>
wrote:

> *@wdoug* commented on this pull request.
> ------------------------------
>
> In server/config.json
> <https://github.com/codefordenver/encorelink/pull/377#discussion_r179339544>
> :
>
> > @@ -1,7 +1,7 @@
>  {
>    "restApiRoot": "/api",
>    "host": "0.0.0.0",
> -  "port": 3000,
> +  "port": 54321,
>
> @nlkluth <https://github.com/nlkluth> is this just arbitrary?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codefordenver/encorelink/pull/377#pullrequestreview-109565042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUU5JvuTchq7bFv9VyzIT0aQAg9GZS4Qks5tlYgfgaJpZM4Sn8Zs>
> .
>



-- 
James Sullivan
Web Developer
(214) 621-0523
jamesfsullivan5@gmail.com
Github <https://github.com/jsullivan5> | Portfolio
<https://jsullivan5.github.io/> | LinkedIn
<https://www.linkedin.com/in/james-f-sullivan/>

wdoug committed:
Fix linting and a few other small things

wdoug committed:
Fix react storybook and directly import images

wdoug committed:
Fix travis CI

wdoug committed:
Fix one last lint error

wdoug committed:
Set up direct support for scss in react storybook

This is done so it doesn't cause issues when the css file hasn't been compiled yet

wdoug committed:
Simplify setup

wdoug committed:
Include updated front-end serving for review apps

wdoug committed:
Fix ability to agree to terms

wdoug committed:
Only check for email variables we actually use

wdoug committed:
Minor fix to emails

wdoug committed:
Update static file server setup for production deploys

wdoug committed:
Utilize heroku-postbuild for building the front-end

wdoug committed:
Connect netlify review apps to heroku review apps

wdoug commented on client/src/containers/AboutContainer.js:
```diff
@@ -1,41 +0,0 @@
-import React from 'react';
```
This file was moved and updated so git thinks it is a new file. See the updated version [here](https://github.com/codefordenver/encorelink/pull/377/files#diff-8bb5f9d03567507880a443ff628d5db2)

wdoug commented on client/src/scss/components/_about.scss:
```diff
@@ -1,4 +0,0 @@
-.about-img img {
```
This file was moved, and it looks like the whitespace changed

wdoug commented on client/stories/__tests__/storyshots.spec.js:
```diff
@@ -1,9 +0,0 @@
-import initStoryshots from 'storyshots';
```
[Updated version of this file in the PR](https://github.com/codefordenver/encorelink/pull/377/files#diff-798b49dc40c590ee34c50a7889bd0ca4)

wdoug commented on client/stories/addons.js:
```diff
@@ -1,3 +0,0 @@
-import '@kadira/storybook/addons';
```
[Updated version of this file in the PR](https://github.com/codefordenver/encorelink/pull/377/files#diff-f5db38d9ba5560598e902c026311b340)

dviramontes reviewed (commented):
At a quick glance, this change-set looks good but I feel like it's too big to review by one person without pulling it down and testing it locally. I suggest we break it up between multiple people over multiple sessions maybe ?

wdoug commented on client/stories/config.js:
```diff
@@ -1,19 +0,0 @@
-import { configure, setAddon } from '@kadira/storybook';
```
[Updated version of this file in the PR](https://github.com/codefordenver/encorelink/pull/377/files#diff-6ab7cd89286451cc2620c7509b7d23b8)

wdoug commented on client/stories/eventrow.js:
```diff
@@ -1,22 +0,0 @@
-import React from 'react';
```
[Updated version of this file](https://github.com/codefordenver/encorelink/pull/377/files#diff-bae5a47d39013a0af5f8ee1c1601d7d1)

wdoug committed:
Pull out Google API key into environment variable

nlkluth committed:
move api call to after initialization

wdoug reviewed (approved):
LGTM
